### PR TITLE
fix up / make consistent themes definitions

### PR DESCRIPTION
**Pull request by tomkralidis**
Follow-on of #237, as per https://github.com/opengeospatial/ogcapi-records/issues/235#issuecomment-1528834725

- cleanup of queryables examples based on core queryables
- minor editorial updates/fixes

I also refactored the themes definition into its own schema fragment/file for reuse.  In theory we could use this in https://github.com/opengeospatial/ogcapi-records/blob/master/core/standard/clause_12_local_resources_catalogue.adoc#example, but the example would then show a schema reference rather than the actual inline definition (which is valuable from a spec readability perspective, but another place to update if things change i the themes definition).

Thoughts @pvretano @m-mohr?

### CODE REVIEW

File: core/examples/json/queryables.json [hunk 1]
```diff
@@ -5,68 +5,68 @@
...
-            "queryable": "publisher",
-            "type": "datetime"
+            "queryable": "description",
+            "type": "string"
         },
         {
-            "queryable": "themes",
-            "type": "datetime"
+            "queryable": "keywords",
+            "type": "string"
```
Comment (m-mohr) [inline core/examples/json/queryables.json:33]: How's the type determined? Isn't that actually an array of strings? Maybe I'm missing something though...

**Comment by m-mohr** [comment]
I guess a reference would be okay and simpilfy the example. Or is there any good reason why in practice someone wouldn't use a ref and reuse existing schemas?

**Comment by tomkralidis** [comment]
@m-mohr I've updated the example to match the queryables schema as well as the various types.

As well:
> I guess a reference would be okay and simpilfy the example. Or is there any good reason why in practice someone wouldn't use a ref and reuse existing schemas?

For an implementation, yes we would ref and reuse existing schemas. The question is whether to use a `$ref` in the example in the specification or be explicit to demonstrate the concept?

**Comment by pvretano** [comment]
01-MAY-2023: Closes #235

**Comment by m-mohr** [comment]
My review is above and did not receive comments yet.

File: core/examples/json/queryables.json (hunk 1 above)
Comment (tomkralidis) [reply core/examples/json/queryables.json]: I've updated the type based on the type of the actual queryable.  For example, if `keywords` is an array, the queryable is of type `array`.  Users searching would pass a string literal (`keywords=birds`).  @pvretano is this the right interpretation of defining queryables?

Comment (m-mohr) [reply core/examples/json/queryables.json]: That makes sense, I've just seen other implementations (in STAC?), where the queryables are 1:1 mappings to the actual "property" (here array). This left me confused. If this PR is the correct way forward, great.

File: core/openapi/schemas/catalogue.yaml
Comment (pvretano) [inline core/openapi/schemas/catalogue.yaml:1]: @tomkralidis the schema for the `themes` member should be:
````
themes:
  type: array
  description: A knowledge organization system used to classify the catalogue.
  items:
    $ref: theme.yaml
````
The PR I recently merged created a `theme.yaml` file that is used in few places.

File: core/openapi/schemas/themes.yaml
Comment (pvretano) [inline core/openapi/schemas/themes.yaml:1]: @tomkralidis remove this file and verify that the [`theme.yaml`](https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/theme.yaml) file that is now in the master branch contains the same information for a single theme.

File: core/openapi/schemas/catalogue.yaml
Comment (tomkralidis) [reply core/openapi/schemas/catalogue.yaml:1]: @pvretano fixed/rebased.

File: core/openapi/schemas/themes.yaml
Comment (tomkralidis) [reply core/openapi/schemas/themes.yaml:1]: FYI I'm now following what is in master, which means `themes.yaml`.

Comment (tomkralidis) [reply core/openapi/schemas/themes.yaml:1]: Wait, I see the following now in master:

https://github.com/opengeospatial/ogcapi-records/blob/master/core/openapi/schemas/recordGeoJSON.yaml#L110

...but there is no `themes.yaml`, only `theme.yaml`.

Summary: I've updated this PR to point to `theme.yaml`.  Should be all aligned now.

**Review by pvretano — APPROVED** [review]
@tomkralidis looks good.  Do we still need the `themes.yaml` file?

**Comment by tomkralidis** [comment]
@pvretano good catch, removed.  Thanks

P.S. if/once this PR is merged, it should be "Squash and Merge"d (lots of superfluous commits).

**Review by m-mohr — APPROVED** [review]
Looks like a good improvement

**Pull request merged**